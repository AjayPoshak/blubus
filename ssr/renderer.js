export const createScriptTag = script => `<script src=${script}></script>`;

const renderFullPage = (html, preloadedState, styles, bundles) => `<!DOCTYPE html>
	<html lang="en">
	
	<head>
		<meta charset="UTF-8">
		<title>Bus Booking in Africa</title>
		<meta http-equiv="X-UA-Compatible" content="ie=edge">
		<meta name="viewport" content="width=device-width, initial-scale=1.0">
		<!-- multiple icons for IE -->
		<meta name="theme-color" content="#2396b0">
		<meta name="msapplication-square310x310logo" content="icon_largetile.png">
		<meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
		<meta name="description" content="online bus booking service in Rwanda, 
		book bus online from anywhere">
		<!-- icon in the highest resolution we need it for -->
		<!-- reuse same icon for Safari -->
		<link rel="manifest" href="../public/manifest.json">
		<link rel="apple-touch-icon" sizes="57x57" href="/apple-icon-57x57.png">
		<link rel="apple-touch-icon" sizes="60x60" href="/apple-icon-60x60.png">
		<link rel="apple-touch-icon" sizes="72x72" href="/apple-icon-72x72.png">
		<link rel="apple-touch-icon" sizes="76x76" href="/apple-icon-76x76.png">
		<link rel="icon" sizes="192x192" href="/public/images/BluBus_192x192.png">
		<link rel="icon" type="image/png" sizes="96x96" href="/favicon-96x96.png">
		<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
		<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
		<link rel="apple-touch-icon" sizes="114x114" href="/apple-icon-114x114.png">
		<link rel="apple-touch-icon" sizes="120x120" href="/apple-icon-120x120.png">
		<link rel="apple-touch-icon" sizes="144x144" href="/apple-icon-144x144.png">
		<link rel="apple-touch-icon" sizes="152x152" href="/apple-icon-152x152.png">
		<link rel="apple-touch-icon" sizes="180x180" href="/apple-icon-180x180.png">
		<link rel="apple-touch-startup-image" href="/public/images/BluBus_192x192.png">
		<link rel="icon" type="image/png" sizes="192x192"  href="/android-icon-192x192.png">
		<meta name="msapplication-TileColor" content="#ffffff">
		<meta name="msapplication-TileImage" content="/ms-icon-144x144.png">
		${styles}
	</head>
	
	<body>
		<div id="root">${html}</div>
	</body>
	<link href="https://fonts.googleapis.com/css?family=Roboto" rel="stylesheet">
	<script src=https://cdn.ravenjs.com/3.24.0/raven.min.js
 crossorigin=anonymous></script>
	<script>
		window.__PRELOADED_STATE__ = ${JSON.stringify(preloadedState).replace(/</g, '\\u003c')}
	</script>
	<script src=${bundles.manifest.js}></script>
	<script src=${bundles.main.js}></script>
	<script src=${bundles.vendors.js}></script>
	<script>
		if('serviceWorker' in navigator) {
			window.addEventListener('load', () => {
				navigator.serviceWorker.register('./service-worker.js')
			})
		}
	</script>
	<!-- Third Party CSS for calendar module -->
	<link rel="stylesheet" href="/public/rc-calendar.css">
	</html>`;
export default renderFullPage;
