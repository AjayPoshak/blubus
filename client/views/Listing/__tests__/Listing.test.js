import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router';
import renderer from 'react-test-renderer';

import { Listing } from '../index';

function mountComponent(props = {}) {
	return mount(
		<MemoryRouter>
			<Listing {...props} />
		</MemoryRouter>
	);
}

describe('Schedules listing component', () => {
	const mockData = {
		listing: {
			items: {
				journeyDetails: { to: 'Kigali', from: 'kibuye', depDate: '12 Jan', arrivalDate: '14 Jan' },
				services: [
					{
						companyName: 'MR Travels',
						busType: 'AC',
						dep: '',
						arr: '',
						travelTime: '',
						ticketPrice: '',
						currency: ''
					},
					{
						companyName: 'MRP Travels',
						busType: 'AC',
						dep: '',
						arr: '',
						travelTime: '',
						ticketPrice: '',
						currency: ''
					}
				]
			}
		},
		dispatch: jest.fn()
	};
	let wrapper;
	beforeEach(() => {
		wrapper = mountComponent(mockData);
	});

	it('rendering journey details', () => {
		expect(
			wrapper
				.find('section header h2')
				.first()
				.text()
		).toBe('kibuye');
	});

	it('rendering journey schedules', () => {
		// Checking if journey details are rendered as many times the data is passed.
		// Rest of rendering of props has been handled in UT of ListingCard.
		expect(wrapper.find('ListingCard').length).toBe(2);
	});

	it('contains header component', () => {
		expect(wrapper.find('Header').length).toBe(1);
	});

	it('UI snapshots', () => {
		const tree = renderer.create(mountComponent(mockData)).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
