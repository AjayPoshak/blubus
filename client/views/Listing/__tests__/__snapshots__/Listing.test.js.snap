// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schedules listing component UI snapshots 1`] = `
<section>
  <section
    className="sc-jTzLTM kmjhrp"
  >
    <a
      href="/"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <header>
        <span
          className="sc-gZMcBi jZgJdZ"
        >
          <img
            alt="blubus logo"
            className="sc-gqjmRU cqEElJ"
            src="../../public/images/BluBus_192x192.png"
          />
          <h1
            className="sc-VigVT beOaog"
          >
            BluBus
          </h1>
        </span>
      </header>
    </a>
  </section>
  <section>
    <header
      className="sc-htpNat kiiLbN"
    >
      <h2>
        kibuye
      </h2>
      <img
        alt="arrow"
        className="sc-bxivhb kIfBUr"
        src="https://res.cloudinary.com/ddbxa4afa/image/upload/v1528049570/blubus/right-arrow.svg"
      />
      <h2>
        Kigali
      </h2>
    </header>
  </section>
  <article
    className="sc-bdVaJa fRGHqF"
  >
    <section
      className="sc-bwzfXH hcNNxv"
      onClick={[Function]}
    >
      <header
        className="sc-ifAKCX bWqqkl"
      >
        <h2
          className="sc-EHOje hlHsbz"
        >
          MR Travels
        </h2>
        <p
          className="sc-bZQynM gUHbKa"
        >
          
           
          
        </p>
      </header>
      <article
        className="sc-gzVnrw kyzxxf"
      >
        <p>
          
        </p>
        <img
          alt="traveling-bus"
          className="sc-dnqmqq cswPpA"
          src="https://res.cloudinary.com/ddbxa4afa/image/upload/v1528048630/blubus/traveling-bus.svg"
        />
        <p>
          
        </p>
        <div
          className="sc-htoDjs blXLnW"
        >
          <img
            alt="timer"
            className="sc-iwsKbI chODOK"
            src="https://res.cloudinary.com/ddbxa4afa/image/upload/v1528048884/blubus/timer.svg"
          />
          <p>
            
             hours
          </p>
        </div>
      </article>
    </section>
    <section
      className="sc-bwzfXH hcNNxv"
      onClick={[Function]}
    >
      <header
        className="sc-ifAKCX bWqqkl"
      >
        <h2
          className="sc-EHOje hlHsbz"
        >
          MRP Travels
        </h2>
        <p
          className="sc-bZQynM gUHbKa"
        >
          
           
          
        </p>
      </header>
      <article
        className="sc-gzVnrw kyzxxf"
      >
        <p>
          
        </p>
        <img
          alt="traveling-bus"
          className="sc-dnqmqq cswPpA"
          src="https://res.cloudinary.com/ddbxa4afa/image/upload/v1528048630/blubus/traveling-bus.svg"
        />
        <p>
          
        </p>
        <div
          className="sc-htoDjs blXLnW"
        >
          <img
            alt="timer"
            className="sc-iwsKbI chODOK"
            src="https://res.cloudinary.com/ddbxa4afa/image/upload/v1528048884/blubus/timer.svg"
          />
          <p>
            
             hours
          </p>
        </div>
      </article>
    </section>
  </article>
</section>
`;
